app/Party.php

```diff
@@ -394,7 +401,15 @@ public function scopeUpcomingEvents($query, $by_event = false)
 
       return $this->join('groups', 'groups.idgroups', '=', 'events.group')
             ->join('users_groups', 'users_groups.group', '=', 'groups.idgroups')
-            ->whereNotNull('events.wordpress_post_id')
+            ->leftJoin('events_users', function($join) {
+                $join->on('events_users.event', '=', 'events.idevents');
+                $join->where('events_users.user', '=', Auth::User()->id);
+            })
+            ->where(function($query) {
+              $query
+                  ->whereNotNull('events.wordpress_post_id')
+                  ->orWhere('events_users.role', '=', Role::HOST);
```

Comment:
If it's restricted to a host, would this mean that admins / network coordinators wouldn't see it here until it is approved?  (Just checking - perhaps that's fine, and they see it elsewhere.)

Reply:
They wouldn't see it unless they'd created it, I don't think, but they'd be seeing it in the list of events to approve.